core/src/event_id.rs

```diff
@@ -149,9 +118,211 @@ fn last4_bytes(buf: &[u8]) -> &[u8] {
+        Builder {
+            state: WithNetwork { bytes },
+        }
+    }
+}
+impl Builder<WithNetwork> {
+    // TODO sort_value should be bytes not str
+    pub fn with_sort_value(mut self, sort_value: &str) -> Builder<WithSortValue> {
+        self.state
+            .bytes
+            .extend(last8_bytes(&sha256_digest(sort_value)));
+        Builder {
+            state: WithSortValue {
+                bytes: self.state.bytes,
+            },
+        }
+    }
+}
+impl Builder<WithSortValue> {
+    pub fn with_min_controller(mut self) -> Builder<WithController> {
+        self.state.bytes.extend(ZEROS_8);
+        Builder {
+            state: WithController {
+                bytes: self.state.bytes,
+            },
+        }
+    }
+    pub fn with_max_controller(mut self) -> Builder<WithController> {
+        self.state.bytes.extend(FFS_8);
+        Builder {
+            state: WithController {
+                bytes: self.state.bytes,
+            },
+        }
+    }
+    pub fn with_controller(mut self, controller: &str) -> Builder<WithController> {
+        self.state
+            .bytes
+            .extend(last8_bytes(&sha256_digest(controller)));
+        Builder {
+            state: WithController {
+                bytes: self.state.bytes,
+            },
+        }
+    }
+}
+impl Builder<WithController> {
+    pub fn with_min_init(mut self) -> Builder<WithInit> {
+        self.state.bytes.extend(ZEROS_4);
+        Builder {
+            state: WithInit {
+                bytes: self.state.bytes,
+            },
+        }
+    }
+    pub fn with_max_init(mut self) -> Builder<WithInit> {
+        self.state.bytes.extend(FFS_4);
+        Builder {
+            state: WithInit {
+                bytes: self.state.bytes,
+            },
+        }
+    }
+    pub fn with_init(mut self, init: &Cid) -> Builder<WithInit> {
+        self.state
+            .bytes
+            .extend(last4_bytes(init.to_bytes().as_slice()));
+        Builder {
+            state: WithInit {
+                bytes: self.state.bytes,
+            },
+        }
+    }
+}
+impl Builder<WithInit> {
+    pub fn with_min_event_height(mut self) -> Builder<WithEventHeight> {
+        let mut event_height_cbor = Encoder::from_memory();
+        event_height_cbor.encode([0]).unwrap();
+        // event_height cbor unsigned int
+        self.state.bytes.extend(event_height_cbor.as_bytes());
+        Builder {
+            state: WithEventHeight {
+                bytes: self.state.bytes,
+            },
+        }
+    }
+    pub fn with_max_event_height(mut self) -> Builder<WithEventHeight> {
+        let mut event_height_cbor = Encoder::from_memory();
+        // TODO: Use a smarter value here instead of u64::MAX to keep the event height length
+        // smaller.
+        event_height_cbor.encode([u64::MAX]).unwrap();
+        // event_height cbor unsigned int
+        self.state.bytes.extend(event_height_cbor.as_bytes());
```

Comment:
The first byte of an `event_height_cbor` will always be 27 or less.
We could use the single byte `b"\xFF"` as a byte that will be grater then a 
`1B FFFFFFFFFFFFFFFF \\ event_height_cbor.encode([u64::MAX])`

`b"\xFF"` is the break stop code in CBOR.